src/lib.rs

```diff
@@ -10,6 +10,10 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+pub mod types {
```

Comment:
This is perhaps stylistic preference (and not a blocker for landing the PR), but is there much value in having "types" as a separate thing in the hierarchy rather than something more flat?

Reply:
Helps with reasoning about dependency directions I think.